src/app/trainee/revalidation-history/revalidation-history.component.html

```diff
@@ -1,81 +1,108 @@
-                    <mat-icon *ngSwitchCase="'Rejected'" aria-hidden="true" class="icon-red">error_outline</mat-icon>
-                    <mat-icon *ngSwitchCase="'Under review'" aria-hidden="true" class="icon-blue">done</mat-icon>
-                    <span class="outcome-text">{{ element.outcome }}</span>
-                </div>
-            </td>
-        </ng-container>
-        <ng-container matColumnDef="gmcSubDueDate">
-            <th mat-header-cell scope="col" *matHeaderCellDef> GMC Submission Due Date </th>
-            <td mat-cell *matCellDef="let element"> {{ element.gmcSubDueDate }}
-            </td>
-        </ng-container>
-        <ng-container matColumnDef="actSubDate">
-            <th mat-header-cell scope="col" *matHeaderCellDef> Actual Submission Date </th>
-            <td mat-cell *matCellDef="let element"> {{ element.actSubDate }} </td>
-        </ng-container>
-        <ng-container matColumnDef="submittedBy">
-            <th mat-header-cell scope="col" *matHeaderCellDef> Submitted By </th>
-            <td mat-cell *matCellDef="let element"> {{ element.submittedBy }} </td>
-        </ng-container>
+<div [ngClass]="{'trainee-history-wrapper mat-elevation-z8': (isHandset$ | async) === false}">
+    <div class="left-nav" *ngIf="(isHandset$ | async) === false">
+        <app-trainee-details></app-trainee-details>
+    </div>
+
+    <mat-divider *ngIf="(isHandset$ | async) === false" [vertical]="true"></mat-divider>
 
-        <ng-container matColumnDef="expandedDetail">
-            <td mat-cell *matCellDef="let element" [attr.colspan]="columnsToDisplay.length">
-                <div class="element-detail" [@detailExpand]="element == expandedElement ? 'expanded' : 'collapsed'">
-                    <div class="element-submission-status">
-                        <strong>Submission Status:</strong> {{element.submissionStatus}}
+    <div class="trainee-container "
+        [ngClass]="{'spacer': (isHandset$ | async) === false, 'mat-elevation-z8': (isHandset$ | async) === true}"
+        *ngIf="revalidationHistory$ | async as revalidationHistory">
+        <table mat-table [dataSource]="revalidationHistory.recommendations" multiTemplateDataRows class="w-100">
+            <caption class="collapsed no-height">Trainee revalidation historical details</caption>
+            <ng-container matColumnDef="recommendation">
+                <th mat-header-cell scope="col" *matHeaderCellDef> Recommendation </th>
+                <td mat-cell *matCellDef="let element"> {{ element.recommendation }} </td>
+            </ng-container>
+            <ng-container matColumnDef="outcome">
+                <th mat-header-cell scope="col" *matHeaderCellDef> Outcome</th>
+                <td mat-cell *matCellDef="let element">
+                    <div class="outcome" [ngSwitch]="element.outcome">
+                        <mat-icon *ngSwitchCase="'Approved'" aria-hidden="true" class="icon-green">done_all
```

Comment:
can these string values mapped to a some sort of status enum so they can be reused?

Reply:
great idea. In data-binding ticket we'll need the possible values from back-end guys as discussed in planning. should I pick up here with guessed values, or will you pick-up with data-binding ticket?

Reply:
either way is fine, i have some questions for the BE team as the json response from the api seems  a bit light [well at least in my eyes]